crates/hotshot/src/demo.rs

```diff
@@ -107,9 +101,9 @@ impl TestableState for SDemoState {
     fn create_random_transaction(
         _state: Option<&Self>,
         _rng: &mut dyn rand::RngCore,
-        padding: u64,
+        _padding: u64,
     ) -> <Self::BlockType as BlockPayload>::Transaction {
-        VIDTransaction(vec![0; padding as usize])
```

Comment:
Why did we get rid of padding? Isn't this used for benchmarking?

Reply:
Good catch! I wanted to avoid the `padding = 0` case, since the VID calculation wouldn't work for an empty vector. And I thought the padding wasn't used so I just removed it, which now I realized wasn't correct. Will update this.

Reply:
> ... the VID calculation wouldn't work for an empty vector.

Sounds like a bug in jellyfish. We can use a single zero-byte instead until it's fixed.

Reply:
I'm going to add a `panic!` if `padding == 0` (I think panic is fine here since it's test code.), then use the given padding rather than fixing the size.

Reply:
This new `panic!` causes `test_success` to fail. (I don't know how you got CI to skip tests in commit https://github.com/EspressoSystems/HotShot/pull/1814/commits/2d51285e4f05033f0e156823310d8dba95622a26 ! I thought you need to prefix the commit message with `[no ci]` to accomplish that.)

At first glance it seems the purpose of `padding` is to append some extra noise to the end of a (randomly generated) tx with the understanding that the original tx already has nonzero size and could be further enlarged via `padding`.

But the impl of `create_random_transaction` in this PR starts with a *zero-size* tx, so the *only* data in the tx is due to `padding`. Thus, if `padding` is zero then the entire tx is zero.

Seems we could solve the problem by picking a nonzero minimum size `s` for this "random" tx and then return a tx with size `s + padding`. Make sense?

Reply:
> > ... the VID calculation wouldn't work for an empty vector.
> 
> Sounds like a bug in jellyfish. We can use a single zero-byte instead until it's fixed.

upstream fix: https://github.com/EspressoSystems/jellyfish/pull/377

Reply:
With this update, `test_success` passes without error logs! I'll update the jellyfish dependency in the cargo file after your PR is merged.

Reply:
Hmm but `test_catchup_web` fails. I'll investigate.

Reply:
The second run passes, so looks like it's the nondeterministic issue we have on main as well.